### PR TITLE
Implement student_email_list on Course

### PR DESCRIPTION
**Pull request by oiratka**

### CODE REVIEW

File: app/models/course.rb [hunk 1]
```diff
@@ -4,4 +4,20 @@ class Course < ApplicationRecord
   has_many :enrollments
   has_many :students, through: :enrollments
```
Comment (inlibra-cpa) [inline app/models/course.rb:5]: Ensure empty course returns `[]`.
Avoid duplicate emails if duplicate enrollments exist. Add `-> { distinct }` to `has_many :students`.

File: spec/models/course_spec.rb [hunk 2]
```diff
@@ -101,7 +101,7 @@
     end
   end
 
-  describe '.student_email_list', skip: true do
+  describe '.student_email_list' do
```
Comment (inlibra-cpa) [inline spec/models/course_spec.rb:104]: Unskip `.student_email_list`, per instructions and add following cases:
- zero-enrollment case;
- duplicate enrollment case (should de-dupe);
- ordering expectation (by email or name, match implementation)

**Review by inlibra-cpa — CHANGES_REQUESTED** [review]
Hi @oiratka ,

Nice start!
Please replace the enrollment loop with association plucks (and distinct) to eliminate N+1, define deterministic ordering, and add tests for empty/duplicate enrollment cases.
